python/ray/air/_internal/torch_utils.py

```diff
@@ -168,11 +169,22 @@ def load_torch_model(
 ) -> torch.nn.Module:
     """Loads a PyTorch model from the provided ``saved_model``.
 
-    If ``saved_model`` is a torch Module, then return it directly. If ``saved_model`` is
-    a torch state dict, then load it in the ``model_definition`` and return the loaded
-    model.
+    ``model_definition`` is used and only used when ``saved_model`` is
+    a torch state dict, which will be loaded into ``model_definition``.
+    Otherwise, ``model_definition`` is discarded.
     """
     if isinstance(saved_model, torch.nn.Module):
+        if model_definition:
+            warnings.warn(
+                "TorchCheckpoint already contains all information needed. "
```

Comment:
Let’s move this warning directly in TorchCheckpoint? Not sure if it makes sense in this util function.

Reply:
@amogkam not sure if you saw this related conversation https://github.com/ray-project/ray/pull/28304#discussion_r970155736